### PR TITLE
DragCaptureZone touch support

### PR DESCRIPTION
**Pull request by ogupte**
Adds touch support for DragCaptureZone and fixes bug in example
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [ ] Firefox
  - [x] Safari
  - [ ] IE11 (Win 7)
  - [ ] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- ~~One core team UX approval~~

### CODE REVIEW

File: src/components/DragCaptureZone/DragCaptureZone.jsx [hunk 1]
```diff
@@ -65,24 +72,40 @@ const DragCaptureZone = createClass({
...
 				onMouseDown={this.handleDragStart}
+				ref={(ref) => {this.elementRef = ref;}}
 			/>
 		);
 	},
 
+	componentDidMount() {
+		//add event listeners directly on the DOM elemnt to allow preventDefault
+		//calls which are not honored due to react's event delegation
+		this.elementRef.addEventListener('touchstart', this.handleDragStart);
```
Comment (jondlm) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: Shouldn't you remove this listeners on unmount?

File: src/components/DragCaptureZone/DragCaptureZone.jsx [hunk 2]
```diff
@@ -45,13 +45,20 @@ const DragCaptureZone = createClass({
 		 * Signature: `({ dx, dy, pageX, pageY }, { event, props }) => {}`
 		 */
 		onDragStart: func,
+		/**
+		 * Called when the drag event is canceled dur to user interaction
```
Comment (chrisdopuch) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: dur -> due
also you could add something about the cancel being triggered by them moving their point of touch off of the screen.

File: src/components/DragCaptureZone/DragCaptureZone.jsx [hunk 3]
```diff
@@ -65,24 +72,40 @@ const DragCaptureZone = createClass({
...
 				className={cx('&', this.props.className)}
 				key='DragCaptureZone'
 				onMouseDown={this.handleDragStart}
+				ref={(ref) => {this.elementRef = ref;}}
 			/>
 		);
 	},
 
+	componentDidMount() {
+		//add event listeners directly on the DOM elemnt to allow preventDefault
```
Comment (chrisdopuch) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: elemnt -> element

File: src/components/DragCaptureZone/DragCaptureZone.jsx [hunk 4]
```diff
@@ -65,24 +72,40 @@ const DragCaptureZone = createClass({
...
+		//calls which are not honored due to react's event delegation
```
Comment (chrisdopuch) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: this is the Chrome only bug you were talking about right? Link to some related reading might be useful as part of the comment or something.

File: src/components/DragCaptureZone/DragCaptureZone.jsx [hunk 5]
```diff
@@ -65,24 +72,40 @@ const DragCaptureZone = createClass({
...
 		window.document.removeEventListener('mousemove', this.handleDrag);
 		window.document.removeEventListener('mouseup', this.handleDragEnd);
 	},
 
 	handleDrag(event) {
-		const {
-			pageX,
-			pageY,
-		} = event;
+		let pageX;
```
Comment (chrisdopuch) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: What do we do with these variables? They just get assigned in this function but not used.

Comment (chrisdopuch) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: Nevermind, didn't notice the rest of the function, carry on!

File: src/components/DragCaptureZone/DragCaptureZone.jsx [hunk 6]
```diff
@@ -150,6 +185,18 @@ const DragCaptureZone = createClass({
 			pageY,
 		});
 	},
+
+	handleDragCancel(event) {
```
Comment (chrisdopuch) [inline src/components/DragCaptureZone/DragCaptureZone.jsx]: Does this event also need `.preventDefault`?

**Pull request merged**